pkg/globalnet/controllers/cluster_egressip_controller.go

```diff
@@ -0,0 +1,117 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controllers
+
+import (
+	"fmt"
+
+	"github.com/submariner-io/admiral/pkg/federate"
+	"github.com/submariner-io/admiral/pkg/syncer"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/globalnet/controllers/ipam"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+func NewClusterGlobalEgressIPController(config syncer.ResourceSyncerConfig, pool *ipam.IPPool) (Interface, error) {
+	var err error
+
+	klog.Info("Creating ClusterGlobalEgressIP controller")
+
+	// TODO - get singleton ClusterGlobalEgressIP and prime the IPPool cache.
+
+	controller := &clusterGlobalEgressIPController{
+		pool:   pool,
+		stopCh: make(chan struct{}),
+	}
+
+	controller.resourceSyncer, err = syncer.NewResourceSyncer(&syncer.ResourceSyncerConfig{
+		Name:                "ClusterGlobalEgressIP syncer",
+		ResourceType:        &submarinerv1.ClusterGlobalEgressIP{},
+		SourceClient:        config.SourceClient,
+		SourceNamespace:     corev1.NamespaceAll,
+		RestMapper:          config.RestMapper,
+		Federator:           federate.NewUpdateStatusFederator(config.SourceClient, config.RestMapper, corev1.NamespaceAll),
+		Scheme:              config.Scheme,
+		Transform:           controller.process,
+		ResourcesEquivalent: syncer.AreSpecsEquivalent,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return controller, nil
+}
+
+func (c *clusterGlobalEgressIPController) Start() error {
+	klog.Info("Starting ClusterGlobalEgressIP controller")
+
+	return c.resourceSyncer.Start(c.stopCh)
+}
+
+func (c *clusterGlobalEgressIPController) Stop() {
+	close(c.stopCh)
+}
+
+func (c *clusterGlobalEgressIPController) process(from runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
+	clusterGlobalEgressIP := from.(*submarinerv1.ClusterGlobalEgressIP)
+
+	klog.Infof("Processing %sd %#v", op, clusterGlobalEgressIP)
+
+	switch op {
+	case syncer.Create:
+		prevStatus := clusterGlobalEgressIP.Status
+		requeue := c.onCreate(clusterGlobalEgressIP)
```

Comment:
Here we should check the name of the object as we only support cluster-default for now.

From the enhancement PR:
```
When Submariner Globalnet is deployed in a Cluster, it auto-allocates the configured number of globalIPs and programs the necessary egress rules on the active Gateway node of the Cluster. This is done internally by creating ClusterGlobalEgressIP CR object with the name cluster-default. Users can query the CR to check the allocated GlobalIPs at the Cluster level. If admin creates additional ClusterGlobalEgressIP CRs that match the Cluster scope, the status field of those CRs will be marked as error with appropriate message. Basically only cluster-default object will be supported.
```
Link: https://github.com/submariner-io/enhancements/blob/devel/submariner/globalnet-enhancement2-0.md

Reply:
Cool. I see that you are already handling it in onCreate method. Nice.